### PR TITLE
Merge master/2.0.5 into 2.1.0

### PR DESCRIPTION
**Pull request by Arceveti**
Moved from #535

### CODE REVIEW

**Review by axollyon — APPROVED** [review]
redo review

**Review by gheskett — APPROVED** [review]

File: src/game/mario_step.c [hunk 1]
```diff
@@ -259,10 +259,9 @@ s32 stationary_ground_step(struct MarioState *m) {
     if (takeStep) {
         stepResult = perform_ground_step(m);
     } else {
-        // Hackersm64: this condition fixes potential downwarps
-        if (m->pos[1] <= m->floorHeight + 160.0f) {
-            m->pos[1] = m->floorHeight;
-        }
+        //! TODO - This is responsible for many stationary downwarps but is
```
Comment (thecozies) [inline src/game/mario_step.c:262]: scary

File: src/game/mario_step.c [hunk 2]
```diff
@@ -404,8 +403,10 @@ struct Surface *check_ledge_grab(struct MarioState *m, struct Surface *prevWall,
     if (ledgeFloor == NULL
         || (*ledgeFloor) == NULL
         || ledgePos[1] < nextPos[1] + 100.0f
+#ifdef DONT_LEDGE_GRAB_STEEP_SLOPES
         || (*ledgeFloor)->normal.y < COS25 // H64 TODO: check if floor is actually slippery
-        || SURFACE_IS_UNSAFE((*ledgeFloor)->type)) {
```
Comment (thecozies) [inline src/game/mario_step.c:408]: disabling qslgs should be a define again soon

**Review by thecozies — APPROVED** [review]
if @gheskett is good with audio stuff then I'm good

File: src/game/mario_step.c (hunk 2 above)
Comment (arthurtilly) [reply src/game/mario_step.c:408]: literally why

who has ever thought "you know what is ruining my hack? qslgs". plus it looks like ass when they're disabled like this, hyena posted a video and it honestly looks far more jank than when they're possible

File: src/game/mario_step.c (hunk 1 above)
Comment (arthurtilly) [reply src/game/mario_step.c:262]: not even scary this is just vanilla 😔 whats scary is massive untested vanilla behavior changes

**Review by arthurtilly — APPROVED** [review]
epic pr fail

File: src/game/mario_step.c (hunk 2 above)
Comment (axollyon) [reply src/game/mario_step.c:408]: image
lol

Comment (arthurtilly) [reply src/game/mario_step.c:408]: still aint gonna be a define get rolled

File: src/game/mario_step.c (hunk 1 above)
Comment (thecozies) [reply src/game/mario_step.c:262]: @arthurtilly this results in jank itself so I think a solution should be found, you don't run into it in vanilla as much as you do in hacks

File: src/game/mario_step.c (hunk 2 above)
Comment (thecozies) [reply src/game/mario_step.c:408]: yup some people don't want them and it's not just me, we had a convo about this the other day in jams or something that it sucks to remove this as an option

File: src/game/mario_step.c (hunk 1 above)
Comment (arthurtilly) [reply src/game/mario_step.c:262]: where do you run into it in hacks? i cant think of any common downwarps

Comment (gheskett) [reply src/game/mario_step.c:262]: I've run into the downwarp multiple different times in DimensiOne. It's actually pretty common.

Comment (arthurtilly) [reply src/game/mario_step.c:262]: that downwarp is different and not really caused by this code, the problem is the action getting changed to a grounded one in the first place when it shouldnt. should be fixed yeah but this isnt a good solution

**Pull request merged**